### DISCUSSION TITLE
Sleep on recorded macro

### DISCUSSION THREAD

**Discussion by mrcruz** [cajhin/capsicain#52] [Ideas]
First of all, I would like to thank you immensely. I have been suffering from hand and finger pains for quite some time and your decision to make your work public available is helping me in reducing my pains and improving my life considerably. I really appreciate you making this open source and available for anyone that might benefit.

Now, to my idea. My use case is the following: I would like to save a file, alt+tab and then reload the page with an F5. The only problem is that the F5 is too fast for the browser to register. I would need a way to set a sleep on a live recorded macro. In my mind, this could be a fixed amount of time, because most of the time you would only need a fraction of a second. maybe a special ESC shortcut that adds a sleep to the recorded macro? that way you could add as much delay as you want, in fixed steps.

I know I can do this in a bunch of different ways without capsicain, but this would cover a bunch of different use cases and make a solid solution for automating many different tools, on the fly. I also  know that I could define in the config, but live recording macro is much more powerful for someone like me that has to deal with multiple different tools on a daily basis.

Again, thanks so much for your time. Also, I plan to contribute with your project in the future once my hands improve a bit and I find something that I can actually do since low level c++ is not my forte.

**Comment by cajhin** [comment]
If that's a fixed macro, just put it into your ini and add a _SLEEP:500_ before F5. 

See https://github.com/cajhin/capsicain/wiki/Function%3A-sequence%28%29

and https://github.com/cajhin/capsicain/wiki/Macros

For slowing down all key sequences (I think this works for recorded macros, too)
[,] and [.]: delay between keys in sequences -/+ 1ms

**Reply by mrcruz** [reply]
somehow I missed that line on the docs. It does work with live macros, so the problem is fixed. Thanks!
